### PR TITLE
[Artur Kostiuk] product order

### PR DESCRIPTION
**Pull request by art1kos**

### CODE REVIEW

**Comment by art1kos** [comment]
https://issues.griddynamics.net/browse/UIT-25

File: src/app/product-details-page/components/product-description/product-order/product-order.component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  ngOnInit() {
+    if (!this.sizes && !this.price) {
+      return;
+    }
+    this.productConfiguration.price = this.price;
+
+  }
+
+  changePlus() {
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-order/product-order.component.ts]: how the plus can be changed?) The name doesn't describe what this function do

File: src/app/product-details-page/components/product-description/product-order/product-order.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    this.productConfiguration.count++;
+  }
+
+  changeMinus() {
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-order/product-order.component.ts]: The name doesn't describe what this function do

**Review by dverbovyi — CHANGES_REQUESTED** [review]
@art1kos looks good.

please fix minor comments

and add add-to-cart button handler with the ability to send the request to add-to-cart/ route with selected product options

POST add-to-cart/<product-id>

File: src/app/product-details-page/components/product-description/product-order/product-order.component.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    price: ''
+  };
+
+  constructor() { }
+
+  ngOnInit() {
+    if (!this.sizes && !this.price) {
+      return;
+    }
+    this.productConfiguration.price = this.price;
```
[the remark this answers is not in the document]
Comment (art1kos) [reply src/app/product-details-page/components/product-description/product-order/product-order.component.ts]: what this comment mean?

[the remark this answers is not in the document]
Comment (dverbovyi) [reply src/app/product-details-page/components/product-description/product-order/product-order.component.ts]: the default property value

File: src/app/product-details-page/components/product-description/product-order/product-order.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ProductOrderComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (dverbovyi) [inline src/app/product-details-page/components/product-description/product-order/product-order.component.spec.ts:25]: add product-order cases (size, amount selection)

**Review by dverbovyi — APPROVED** [review]
@art1kos good, I'm merging it, please provide missed tests for product count and size selection

**Pull request merged**